lib/src/supabase_auth.dart

```diff
@@ -303,12 +318,92 @@ extension GoTrueClientSignInProvider on GoTrueClient {
       scopes: scopes,
       queryParams: queryParams,
     );
-    final url = Uri.parse(res.url!);
-    final result = await launchUrl(
-      url,
-      mode: authScreenLaunchMode,
-      webOnlyWindowName: '_self',
+    final uri = Uri.parse(res.url!);
+
+    final willOpenWebview = authScreenLaunchMode == LaunchMode.inAppWebView &&
+        context != null &&
+        !kIsWeb && // `Platform.isIOS` throws on web, so adding a guard for web here.
+        Platform.isIOS;
+
+    if (willOpenWebview) {
```

Comment:
Updated the code so that webview will only launch on iOS as Android does not allow performing OAuth on webviwe.

https://developers.googleblog.com/2016/08/modernizing-oauth-interactions-in-native-apps.html

Reply:
Well, I have tested that, and with the current code that this is not working on Android,
we got "Access blocked Error" but indeed when I allow opening WebBrowser on Android, WebBrowser is opened and login works fine.

Reply:
@kkalisz 
Could you give more details on this `Access blocked Error`? What provider did you use and if there are more error message, providing them would be helpful.

Reply:
Hi @dshukertjr,

I'm experiencing exactly the same problem as in https://github.com/supabase/supabase-flutter/issues/357 
So google provider, and 403 disallowed error.
It looks like with the current code, this is expected because for android we are using default inApp browser.

I little bit changed the implementation locally, and it works correctly now. I will try to open PR Today
Thanks for your interest